src/main/java/mate/academy/springboot/datajpa/controller/ProductController.java

```diff
@@ -0,0 +1,70 @@
+package mate.academy.springboot.datajpa.controller;
+
+import java.math.BigDecimal;
+import java.util.List;
+import java.util.Set;
+import java.util.stream.Collectors;
+import lombok.AllArgsConstructor;
+import mate.academy.springboot.datajpa.dto.ProductRequestDto;
+import mate.academy.springboot.datajpa.dto.ProductResponseDto;
+import mate.academy.springboot.datajpa.model.Product;
+import mate.academy.springboot.datajpa.service.ProductService;
+import mate.academy.springboot.datajpa.service.mapper.DtoMapper;
+import org.springframework.http.HttpStatus;
+import org.springframework.web.bind.annotation.DeleteMapping;
+import org.springframework.web.bind.annotation.GetMapping;
+import org.springframework.web.bind.annotation.PatchMapping;
+import org.springframework.web.bind.annotation.PathVariable;
+import org.springframework.web.bind.annotation.PostMapping;
+import org.springframework.web.bind.annotation.RequestBody;
+import org.springframework.web.bind.annotation.RequestMapping;
+import org.springframework.web.bind.annotation.RequestParam;
+import org.springframework.web.bind.annotation.ResponseStatus;
+import org.springframework.web.bind.annotation.RestController;
+
+@RestController
+@AllArgsConstructor
+@RequestMapping("/products")
+public class ProductController {
+    private final ProductService productService;
+    private final DtoMapper<ProductResponseDto, Product, ProductRequestDto> dtoMapper;
+
+    @PostMapping
+    public ProductResponseDto create(@RequestBody ProductRequestDto productRequestDto) {
+        return dtoMapper.toDto(
+                productService.create(dtoMapper.toEntity(productRequestDto)));
+    }
+
+    @GetMapping("/{id}")
+    public ProductResponseDto get(@PathVariable Long id) {
+        return dtoMapper.toDto(productService.findById(id));
+    }
+
+    @PatchMapping("/{id}")
+    public ProductResponseDto update(@PathVariable Long id,
+                                     @RequestBody ProductRequestDto productRequestDto) {
+        return dtoMapper.toDto(productService.update(id,
+                dtoMapper.toEntity(productRequestDto)));
+    }
+
+    @DeleteMapping("/{id}")
+    @ResponseStatus(HttpStatus.OK)
+    public void delete(@PathVariable Long id) {
+        productService.deleteById(id);
+    }
+
+    @GetMapping("/by-price")
+    public List<ProductResponseDto> getAllByPriceBetween(@RequestParam BigDecimal priceFrom,
+                                                         @RequestParam BigDecimal priceTo) {
+        return productService.getAllByPriceBetween(priceFrom, priceTo).stream()
+                .map(dtoMapper::toDto)
+                .collect(Collectors.toList());
```

Comment:
You can use short option
```suggestion
                .toList();
```